front/app/modules/commercial/granular_permissions/admin/containers/Granular/Continuous.tsx

```diff
@@ -33,41 +35,43 @@ interface InputProps {
 interface DataProps {
   permissions: GetProjectPermissionsChildProps;
 }
+
 interface Props extends InputProps, DataProps {}
 
-class Continuous extends PureComponent<Props> {
-  handlePermissionChange = (
+const Continuous = ({ permissions, projectId }: Props) => {
+  const project = useProject({ projectId });
+
+  const handlePermissionChange = (
     permission: IPermissionData,
     permittedBy: IPermissionData['attributes']['permitted_by'],
     groupIds: string[]
   ) => {
     updateProjectPermission(
       permission.id,
-      this.props.projectId,
+      projectId,
       permission.attributes.action,
       { permitted_by: permittedBy, group_ids: groupIds }
     );
   };
 
-  render() {
-    const { permissions, projectId } = this.props;
+  if (!isNilOrError(project)) {
```

Comment:
We can avoid a double nesting here by doing something like this

```
  const config: ParticipationMethodConfig | null = !isNilOrError(project) ? getMethodConfig(project.attributes.participation_method) : null;

  if (!isNilOrError(permissions) && !isNilOrError(config)) {
    return (
      <Container>
        <ActionsForm
          permissions={permissions}
          onChange={handlePermissionChange}
          postType={config.getPostType()}
          projectId={projectId}
        />
      </Container>
    );
  }

  return null;
  ```

Reply:
That's much cleaner - thanks :)